pkg/webui/console/store/middleware/logics/events.js

```diff
@@ -131,22 +129,18 @@ const createEventsConnectLogics = (reducerName, entityName, onEventsStart) => {
         const filterRegExp = Boolean(filter) ? filter.filterRegExp : undefined
 
         try {
-          channel = await onEventsStart([id], filterRegExp, EVENT_TAIL, after)
+          const listeners = {
+            message: message => dispatch(getEventSuccess(id, message)),
+            error: error => dispatch(getEventFailure(id, error)),
+            close: wasClientRequest => dispatch(closeEvents(id, { silent: wasClientRequest })),
+          }
+          channel = await onEventsStart([id], filterRegExp, EVENT_TAIL, after, listeners)
           dispatch(startEventsSuccess(id, { silent }))
-
-          channel.on('message', message => dispatch(getEventSuccess(id, message)))
-          channel.on('error', error => dispatch(getEventFailure(id, error)))
-          channel.on('close', wasClientRequest => {
-            dispatch(closeEvents(id, { silent: wasClientRequest }))
-            channel = null
-          })
+          channel.open()
         } catch (error) {
-          if (isUnauthenticatedError(error)) {
+          if (error instanceof TokenError) {
```

Comment:
We don't really have access to the backend error while using `WebSocket`, but we know that we use the 'latest' token every time. So if something bad happened during token refresh, we have a `TokenError`.